### PR TITLE
Slack plugin

### PR DESCRIPTION
**Pull request by richardguerre**
# Overview

TL;DR, adds the Slack plugin for Flow, allowing users to:
- Connect their Slack workspaces/teams from the settings view
- Add the "Post your plan to Slack" routine step to their routines
- Configure the template message (using Handlebars expressions) and default channels for each routine step added to their routines.
  - If in the `#tasks` context, the `{{slack-status}}` handlebars expression will add the `<slack-status>` tag (which will then be handled by the TipTap extension.
- When doing the "Post your plan to Slack" routine step, the message is crafted from the template and the channels picker has the default channels picked already.
- Clicking on next/done in the routine step, posts the message in Slack.
- Updating the status of any task, will update the message(s) posted in Slack with ✅ or ❌ when either the task is completed or canceled (respectively).

# Cool things

- Using Handlebars to make templates.
  - The syntax and logic is great for what I'm trying to do and I don't have to invent a new language + docs for users.
  - Docs already exist. I'm just missing docs for what helpers/data the templates can use.
- Will potentially use the Monaco editor for the TemplateEditor component so there is correct auto-complete and highlighting ([Monaco for React](https://github.com/suren-atoyan/monaco-react))

# Annoying things

1. The `html-to-slack` library depends on a different html parser than the one used in the server, so there are 2 HTML parsers bundled into the server build.
2. Building the slack plugin with `html-to-slack` results in the keyword `write` showing up in the built javascript file, which gets flagged when installing the plugin. Hence, the `html-to-slack` is currently owned by the server (not the plugin), and is given to plugins as options.

### CODE REVIEW

File: apps/web/src/views/BrowsePluginsView.tsx [hunk 1]
```diff
@@ -88,6 +88,16 @@ const PLUGINS: Plugin[] = [
     version: "0.1.0",
     authors: [{ name: "Flow", avatarUrl: "FlowIcon.svg" }],
   },
+  {
+    iconUrl: "https://slack.com/favicon.ico",
+    name: "Slack",
+    slug: "slack",
+    description:
+      "Official Slack plugin for Flow, allowing you to post your plan to Slack channels when doing your routines.",
+    installUrl: "https://cdn.jsdelivr.net/gh/richardguerre/flow@main/plugins/slack/out",
```
Comment (richardguerre) [inline apps/web/src/views/BrowsePluginsView.tsx:97]: I'll need to modify this URL after merging.
